### PR TITLE
Switch TabBarView to PageView

### PR DESCRIPTION
**Pull request by abarth**
Tabs are now fully driven by slivers.

### CODE REVIEW

**Comment by abarth** [comment]
@Hixie @HansMuller

File: packages/flutter/test/material/tabs_test.dart [hunk 1]
```diff
@@ -389,36 +387,22 @@ void main() {
...
+    await tester.pumpUntilNoTransientCallbacks();
     expect(value, 'RIGHT');
 
     Point rightFlingStart = tester.getCenter(find.text('RIGHT CHILD'));
     await tester.flingFrom(rightFlingStart, const Offset(200.0, 0.0), 10000.0);
-    await tester.pump(); // start the animation
-    await tester.pump(const Duration(milliseconds: 500));
-    await tester.pump(const Duration(milliseconds: 500));
-    await tester.pump(const Duration(milliseconds: 500));
+    await tester.pumpUntilNoTransientCallbacks();
```
Comment (HansMuller) [inline packages/flutter/test/material/tabs_test.dart]: NICE. This is so much better, every time.

**Comment by HansMuller** [comment]
LGTM

File: packages/flutter/test/material/tabs_test.dart (hunk 1 above)
Comment (abarth) [inline packages/flutter/test/material/tabs_test.dart]: :)

**Comment by Hixie** [comment]
Looks like this one also did some regressioning. complex_layout_scroll_perf_ios__timeline: average_frame_build_time_millis, for example, doubled.

**Comment by abarth** [comment]
Great!  Such a huge regression should be easy to find.

**Comment by Hixie** [comment]
Looks like it also slightly regressed microbenchmarks:stock_build_iteration, from ~845ms to ~880ms.

**Pull request merged**